crates/forge/bin/cmd/clone.rs

```diff
@@ -0,0 +1,674 @@
+        if enable_git {
+            let git = Git::new(&root).quiet(true);
+            git.add(Some("--all"))?;
+            git.commit("chore: forge clone")?;
+        }
+
+        Ok(())
+    }
+}
+
+/// Update the configuration file with the metadata.
+/// This function will update the configuration file with the metadata from the contract.
+/// It will update the following fields:
+/// - `auto_detect_solc` to `false`
+/// - `solc_version` to the value from the metadata
+/// - `evm_version` to the value from the metadata
+/// - `via_ir` to the value from the metadata
+/// - `libraries` to the value from the metadata
+/// - `metadata` to the value from the metadata
+///     - `cbor_metadata`, `use_literal_content`, and `bytecode_hash`
+/// - `optimizer` to the value from the metadata
+/// - `optimizer_runs` to the value from the metadata
+/// - `optimizer_details` to the value from the metadata
+///     - `yul_details`, `yul`, etc.
+///     - `simpleCounterForLoopUncheckedIncrement` is ignored for now
+/// - `remappings` and `stop_after` are pre-validated to be empty and None, respectively
+/// - `model_checker`, `debug`, and `output_selection` are ignored for now
+///
+/// Detailed information can be found from the following link:
+/// - https://github.com/foundry-rs/foundry/blob/master/crates/config/README.md#all-options
+/// - https://docs.soliditylang.org/en/latest/using-the-compiler.html#compiler-input-and-output-json-description
+fn update_config_by_metadata(
+    config: &Config,
+    doc: &mut toml_edit::Document,
+    meta: &Metadata,
+    chain: Chain,
+) -> Result<()> {
+    let profile = config.profile.as_str().as_str();
+
+    // macro to update the config if the value exists
+    macro_rules! update_if_needed {
+        ([$($key:expr),+], $value:expr) => {
+            {
+                if let Some(value) = $value {
+                    let mut current = &mut doc[Config::PROFILE_SECTION][profile];
+                    $(
+                        if let Some(nested_doc) = current.get_mut(&$key) {
+                            current = nested_doc;
+                        } else {
+                            return Err(eyre::eyre!("cannot find the key: {}", $key));
+                        }
+                    )+
+                    *current = toml_edit::value(value);
+                }
+            }
+        };
+    }
+
+    // update the chain id
+    doc[Config::PROFILE_SECTION][profile]["chain_id"] = toml_edit::value(chain.id() as i64);
+
+    // disable auto detect solc and set the solc version
+    doc[Config::PROFILE_SECTION][profile]["auto_detect_solc"] = toml_edit::value(false);
+    let version = meta.compiler_version()?;
+    doc[Config::PROFILE_SECTION][profile]["solc_version"] =
+        toml_edit::value(format!("{}.{}.{}", version.major, version.minor, version.patch));
+
+    // get optimizer settings
+    // XXX (ZZ): we ignore `model_checker`, `debug`, and `output_selection` for now,
```

Comment:
You should be able to use [`Metadata::Settings`](https://docs.rs/foundry-block-explorers/latest/foundry_block_explorers/contract/struct.Metadata.html#method.settings) then assign the fields back into the config, instead of manually going through each field.

Reply:
Hi! Thanks for the suggestion! It's greatly appreciated. 

Currently, we utilize `Metadata::settings()` to retrieve the values for each individual field. I'm thinking to implementing a function that serves as the opposite of `Config::solc_settings`. Do you think that would be an appropriate addition?

Reply:
@DaniPopes A second thought. Using settings to update config seems to be a unique use case of `forge clone` and possibly out of scope for a generic function for Config. Might it be advisable to stick with our current approach?